src/foam/u2/ActionView.js

```diff
@@ -401,17 +416,35 @@ foam.CLASS({
       this.addClass(this.myClass(this.action.name));
     },
 
-    function addContent() {
+    async function addContent() {
       /** Add text or icon to button. **/
-      if ( this.icon ) {
-        this.start('img')
-          .style({ 'margin-right': this.label ? '4px' : 0 })
-          .attr('src', this.icon$)
-        .end();
+      var self = this;
+      var size = this.buttonStyle == this.buttonStyle.LINK ? '1em' : this.size.iconSize;
+      var iconStyle = { width: size, height: size, 'margin-right': this.label ? '4px' : '' }; 
+      
+      if ( this.themeIcon ) {
+        var indicator = this.themeIcon.clone(this).expandSVG();
+        this.start(this.HTMLView, { data: indicator }).attrs({ role: 'presentation' }).style(iconStyle).end();
+      } else if ( this.icon ) {
+        if ( this.icon.endsWith('.svg') ) {
+          var req  = this.HTTPRequest.create({
```

Comment:
This seems a little overkill. Is it possible to just use an `img` tag for this case? (using HTMLView from a server response makes a failure mode have less predictable consequences)

Reply:
the goal was to have all SVGs load as inline SVGs so that we could control the styling using CSS. If we are happy with restricting that to only `themeGlyphs` then we could load them in using `img` tags. We'll just need to make sure all Nanopay buttons use the `themeGlyphs`

Reply:
Right, I see the dilemma here - we might sometimes have glyphs that aren't "generic enough" to be on a theme. We could eventually accomplish this using a glyphDAO that gets populated by the filesystem (similarly to FLOW docs) - although this intermediate strategy using HTMLView with a network response is dangerous, since there's nothing stopping the url from linking to somewhere that does a text/html response. It sucks because I see what you did here and it's really cool, but a few more steps are needed to do it securely.